pkg/sidecar/swappable/swappable.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package swappable
+
+import (
+	"errors"
+	"github.com/knative/eventing/pkg/sidecar/multichannelfanout"
+	"go.uber.org/zap"
+	"net/http"
+	"sync"
+	"sync/atomic"
+)
+
+// http.Handler that atomically swaps between underlying handlers.
```

Comment:
Have you considered inverting this to have an SwappableConfig which hands a snapshotted config each time GetConfig is called?

It feels strange to implement HTTPHandler rather than have this be a primitive.

Reply:
If this were inverted, would GetConfig() be called during every HTTP request?

Reply:
The structure of this package is similar to the examples given in [the `atomic.Value` docs](https://golang.org/pkg/sync/atomic/#Value). I'm fine with either style as long as there's no read lock involved.